### PR TITLE
Add num students dropped, num with work, and course name to OSA stats

### PR DESCRIPTION
**Pull request by jpslav**

### CODE REVIEW

File: app/routines/push_salesforce_course_stats.rb [hunk 1]
```diff
@@ -187,18 +187,28 @@ def push_stats(course, os_ancillary)
...
         os_ancillary.num_students += 1
         os_ancillary.num_students_paid += 1 if student.is_paid
         os_ancillary.num_students_comped += 1 if student.is_comped
         os_ancillary.num_students_refunded += 1 if student.first_paid_at.present? && !student.is_paid
+        os_ancillary.num_students_dropped += 1 if student.deleted?
+
+        num_steps_completed = student.role.taskings.map{ |tasking|
+          tasking.task.completed_steps_count
+        }.sum
+
```
Comment (nathanstitt) [inline app/routines/push_salesforce_course_stats.rb]: fwiw sum accepts a block, could use:

```ruby
 num_steps_completed = student.role.taskings.sum{ |tasking| tasking.task.completed_steps_count }
```

File: app/routines/push_salesforce_course_stats.rb [hunk 2]
```diff
@@ -187,18 +187,28 @@ def push_stats(course, os_ancillary)
...
+        os_ancillary.num_students_with_work += 1 if num_steps_completed >= 10
       end
```
Comment (nathanstitt) [inline app/routines/push_salesforce_course_stats.rb]: what's the significance of completing at least more than 10 steps ?  Is that just an internal threshold or something?  If it's going to be changed maybe we should make it a constant or something

File: app/routines/push_salesforce_course_stats.rb (hunk 1 above)
Comment (jpslav) [inline app/routines/push_salesforce_course_stats.rb]: really thought I tried that and it kaboomed on me

File: app/routines/push_salesforce_course_stats.rb (hunk 2 above)
Comment (jpslav) [inline app/routines/push_salesforce_course_stats.rb]: grimaldi requested this as a threshold to figure out how many students are "real".  first time he asks me to change it I'll make it a rails-settings-ui config option :-)

File: app/routines/push_salesforce_course_stats.rb (hunk 1 above)
Comment (nathanstitt) [inline app/routines/push_salesforce_course_stats.rb]: To test I just did:  
```ruby
[{a: 1},{a: 2},{a: 3}].sum{|x| x[:a] }
```

Comment (nathanstitt) [inline app/routines/push_salesforce_course_stats.rb]: although I suspect that Ruby probably does map/sum under the hood anyways so that's fine

**Pull request merged**